pkg/server/middleware/instrument.go

```diff
@@ -34,7 +34,7 @@ var (
 	DefBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 25, 50, 100}
 )
 
-func NewInstrument(routeMatcher RouteMatcher, defBuckets []float64, prefix string) (*Instrument, error) {
+func NewInstrument(routeMatcher RouteMatcher, reg prometheus.Registerer, defBuckets []float64, prefix string) (*Instrument, error) {
```

Comment:
This change seemed necessary for testing, using `prometheus.DefaultRegisterer` caused tests to fail upon re-registration. This allows a NewRegistry() to be passed by the test - but I'm not sure if this change is correct.

Reply:
Fixed by resetting DefaultRegisterer within test teardown